lib/openhab/dsl/rules/builder.rb

```diff
@@ -964,33 +964,68 @@ def every(value, at: nil, attach: nil)
         #
         # Run this rule when the script is loaded.
         #
-        # Execute the rule on OpenHAB start up and whenever the script is
-        # reloaded. This is useful to perform initialization routines,
-        # especially when combined with other triggers.
+        # Execute the rule whenever the script is first loaded, including on OpenHAB start up,
+        # and on subsequent reloads on file modifications.
+        # This is useful to perform initialization routines, especially when combined with other triggers.
         #
-        # @param [true, false] run_on_start Run this rule on start, defaults to True
-        # @param [Object] attach object to be attached to the trigger
+        # @param [Object] attach Object to be attached to the trigger
         # @return [void]
         #
         # @example
         #   rule "startup rule" do
-        #     on_start
+        #     on_load
         #     run do
         #       <calculate some item state>
         #     end
         #   end
         #
         # @example
-        #   rule 'Ensure all security lights are on' do
-        #     on_start
+        #   rule "Ensure all security lights are on" do
+        #     on_load
         #     run { Security_Lights.on }
         #   end
         #
-        # rubocop:disable Style/OptionalBooleanParameter
-        def on_start(run_on_start = true, attach: nil)
-          @on_start = Struct.new(:enabled, :attach).new(run_on_start, attach)
+        def on_load(attach: nil)
+          raise ArgumentError, "on_load can only be called once" if @on_load # prevent overwriting the attachment
+
+          @on_load = { attachment: attach }
+        end
+
+        #
+        # Run this rule when OpenHAB reached a certain start level
+        #
+        # This will only trigger during OpenHAB start up. It won't trigger on script reloads.
+        #
+        # @param [Integer,:rules,:ruleengine,:ui,:things,:complete] at_level Start level between 40 and 100 to trigger.
+        #   | Symbol      | Start Level |
+        #   | ----------- | ----------- |
+        #   | :rules      | 40          |
+        #   | :ruleengine | 50          |
+        #   | :ui         | 70          |
+        #   | :things     | 80          |
+        #   | :complete   | 100         |
+        # @param [Object] attach Object to be attached to the trigger
+        # @return [void]
+        #
+        def on_start(at_level: 100, attach: nil)
+          if at_level.is_a?(Symbol)
+            begin
+              at_level = org.openhab.core.service.StartLevelService.const_get("STARTLEVEL_#{at_level.to_s.upcase}")
+            rescue
+              raise ArgumentError, "Invalid symbol for at_level"
+            end
+          end
+
+          raise ArgumentError, "at_level '#{at_level}' not within (40..100)" unless at_level&.between?(40, 100)
```

Comment:
perhaps allow < 40, but log a warning like "Rule engine doesn't start until start level 40. Rule will not trigger until then"?

Reply:
s/Rule will not trigger until then/Rule will not trigger at all./

https://github.com/openhab/openhab-core/blob/61bb5b3a31af0c594555af8098fe073e783eaaa3/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/SystemTriggerHandler.java#L96

I will allow it anyway, and let core decides what to do, in case the implementation ever changed.

Reply:
> since otherwise the rule engine takes care of the execution already

https://github.com/openhab/openhab-core/blob/61bb5b3a31af0c594555af8098fe073e783eaaa3/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleEngineImpl.java#L1451

of course, this presume the rule is registered before it gets to that start level. this really seems like a bug to me that a rule doesn't fire immediately if it has a start level trigger <= the current start level